app/services/puppet_fact_importer.rb

```diff
@@ -1,9 +1,9 @@
 class PuppetFactImporter < StructuredFactImporter
   def self.authorized_smart_proxy_features
-    'Puppet'
+    'Puppet'.freeze
   end
 
   def fact_name_class
-    ::FactName
+    PuppetFactName
```

Comment:
is there a fact in any plugin that does not use it's own class? I think all existing fact name with type `::FactName` should become `PuppetFactName`